### ISSUE TITLE
Suggestion: Classic Template needs a facelift

### ISSUE THREAD

**Issue by DivaVocals** [zencart/zc-v1-series#311]
*Also being carried over from #304*

Ahhhhh yes..  the proverbial elephant in the room :laughing: I see that "Old Green" is STILL hanging on in v1.6.. :laughing: 

I don't think anyone would disagree that the Classic template is a LOT overdue for a MAKEOVER.. The Classic template that comes with Zen Cart out the box doesn't do a very good job of representing the product IMHO. Now I realize that for years, the emphasis has been on security, and functionality.. and no doubt these are important considerations, but WHY NOT make moves to make Zen Cart LOOK attractive as well.. I typically replace ALL of the Classic template folders and files with the Barbones template's folders and files because while Barebones is PLAIN, it's a LOT nicer looking than Classic.. Is there ANY opposition to a pull request to make Classic at least a LITTLE more visually appealing or has one just not been submitted??

Would like to suggest that It's time to at LEAST let "Old Green" go be with her cousin the "Flaming Gif".. :laughing: 

![Flaming GIFs](https://33.media.tumblr.com/c39718b55fb12187c5862e34bac237da/tumblr_mo3560Ooyd1rx9lqto1_500.gif)

**Comment by drbyte** [comment]
@DivaVocals  said:
> Is there ANY opposition to a pull request to make Classic at least a LITTLE more visually appealing

No, as long as the changes are as SIMPLE as possible. Preferably CSS only. 

I'd be open to removing the 1 remaining layout table as long as the div replacements are compatible with all modern browsers.

Don't go so far as to rewrite it all to HTML5. That's already in the works offline.

**Comment by wiztechinc** [comment]
a can of worms

shall we do some comps and get consensus approval?
ᐧ

Delia Wilson Lunsford
WizTech, Inc., (formerly Delia Wilson Design, LLC.)
434-202-4307
Terms and Conditions for working with WizTech, Inc.
(http://www.teamwiztech.com/terms-conditions.php)

**Comment by lhungil** [comment]
Any thoughts about expanding the "parent" and "child" relationships in future versions of Zen Cart? Right now we have every theme as a child of the "template_default". If this was expanded we could have a parent theme based on "Foundation", "Bootstrap", or any other framework.

This thought has been percolating in the back of my head for quite awhile. Perhaps while looking at updating the generated HTML and starter theme would be a good time to think about (and possibly implement something along these lines)?

**Comment by DivaVocals** [comment]
Folks.. Let me be CLEAR before we all go too far down the rabbit hole here.. what I am propose to do to change Classic is merely a simple stylesheet change and removal of the ugly green images.. Barebones is JUST that.. a simple "barebones" template which removes all those UGLY green images and at least makes an out the box Zen Cart installation a lot more attractive.. Hardly a can of worms.. :smile: 

>Any thoughts about expanding the "parent" and "child" relationships in future versions of Zen Cart? Right now we have every theme as a child of the "template_default". If this was expanded we could have a parent theme based on "Foundation", "Bootstrap", or any other framework.

> This thought has been percolating in the back of my head for quite awhile. Perhaps while looking at updating the generated HTML and starter theme would be a good time to think about (and possibly implement something along these lines)?

I don't disagree that this should be the long term goal, but will the execution of such a change be feasible based on what I understand to be the timeframe for the v1.6 release date.. (sometime this spring/summer I think) This seems like a pretty BIG change..

**Comment by DivaVocals** [comment]
Barebones preview so everyone is CRYSTAL clear on what look I am proposing that classic be updated with. Barebones is simple, clean and a lot more attractive than "old green"

![Image of Barebones](http://www.rowanw.com/screens/barebones_screenshot.png) 
![Image of Classic](http://www.murphsmilitaria.com/images/scr_template_default.jpg)

**Comment by scottcwilson** [comment]
One suggestion I made to @drbyte was to just rename the classic template to (say) "Training".  To me, the purpose of this template is really to show the various divs and spans (in loud garish colors!) so people get a sense of how to build their own template - not so much to be used as-is on a real cart. I do like the idea of including Barebones, which looks nice from the preview above.

**Comment by wiztechinc** [comment]
I definitely agree that a less garish template is the way to go. Classic
does help to emphasize divs to some degree but it's just plain ugly folks.
To someone looking for a website cart it's a hurdle.The barebones look is
much better. Then clients will be saying, "Oh, it's so plain. How can I
fancy it up?" Instead of "oh that's ugly!" and then they go out and buy one
of those god awful templates.

But I want to go a step further. I no longer base my website templates off
of the default classic template. Not everything I do is as easily
adjustable as the classic but there are certain things I do and don't do.

One, I have never ever retained the mainNavWrapper div up at the top. I
haven't done that in more years than I can count. So I never ever use the
default header. I now always have the cart in the header (with a admin
off/on switch). I combine the ezpages nav and the main nav with an option
to have the shopping cart in that bar.

Two, my present approach takes the wider monitor screens into account. I
use visual elements for both header and footer which means in the end I do
not use the base template as that clearly defined width in the middle. This
mimics the bootstrap default look in some ways in an easier fashion so that
both header and footer extend full width while maintaining the rest of the
template in a fixed width.

To do this most effectively, both the headerWrapper and footer wrapper are
removed from the mainWrapper and set up with a fixed width and centered.

Three, I've changed a few elements so that they are more easily styled with
CSS instead of having to install mods such as new category sideboxes.

I'm not saying any of this is a must. I'm saying let's look at ways to make
the default template more easily changeable and more up to date. We don't
have to get fancy but something that can more easily fancied up and made to
look miles from the default.

This is an simple example
[image: Inline image 1]

Here's is that sample template just styled differently:

[image: Inline image 2]

There are a number of examples at deliatest.com/1_5_3.  I also explain my
reasoning for the design and include some new or changed mods that make my
job easier - delivering design / functionality to today's real cart owners.
I'm not trying to sell y'all on all of it. I'm just saying let's look
beyond the look into reformulating the base template.
ᐧ

Delia Wilson Lunsford
WizTech, Inc., (formerly Delia Wilson Design, LLC.)
434-202-4307
Terms and Conditions for working with WizTech, Inc.
(http://www.teamwiztech.com/terms-conditions.php)

**Comment by drbyte** [comment]
@lhungil said
> Any thoughts about expanding the "parent" and "child" relationships in future versions of Zen Cart? Right now we have every theme as a child of the "template_default". If this was expanded we could have a parent theme based on "Foundation", "Bootstrap", or any other framework.

> This thought has been percolating in the back of my head for quite awhile.

We've been toying with this idea too. But as far as the subject of this discussion "Classic template needs a facelift" I think it's off-topic. Discussing it in a separate issue with discussion specifically geared around architectural considerations is probably best, lest it devolve into multiple other topics.

**Comment by IronLady** [comment]
The classic template really need a serious make over. I agree with that. 
But I am not agree if we use barebones to replace classic. I think it's still looks not nice.
What we need is default template that looks fine/good so we can attract "new user" to try and maybe use ZC.
By using css framework, customize the color a bit, categories drop down menu, add feature to create banner slider by default (Who doesn't need drop down menu and slider these days). I think it's enough for store owner to create a modern looking store using ZC.

**Comment by lhungil** [comment]
@drbyte & @DivaVocals
>Discussing it in a separate issue with discussion specifically geared around architectural considerations is probably best, lest it devolve into multiple other topics.

Agreed.

@IronLady 
> What we need is default template that looks fine or maybe good, so we can attract "new user" to try and maybe use ZC.

Agreed. Although depending on the timeline for release, we may end up needing to start with one of the existing "basic" themes as the default template (for example barebones) . Then a new "classic" theme could be added which looks snazzy.

**Comment by DivaVocals** [comment]
@lhungil 
>Agreed. Although depending on the timeline for release, we may end up needing to start with one of the existing "basic" themes as the default template (for example barebones) . 

And THAT ladies and gentlemen is ALL I am suggesting.. A small BEGINNING step.. One that needs to happen sooner than later.. Why WAIT to facelift Classic while the discussion of what to do with the default template is FAR from decided.. This is an EASY yes decision, and Barebones will give everyone a clean start.. 

@scottcwilson I don't dispute that Classic is a starting place, but it's an UGLY starting place, and to the point @wiztechinc makes for a new shopowner who knows nothing about CSS this only sends them to places like Theme Forest or Template Monstrosity..

**Comment by drbyte** [comment]
> Classic is a starting place, but it's an UGLY starting place

Okay, on the issue of facelifting classic (the title of this discussion), all I'm seeing is everyone saying that Classic is ugly and needs to be changed. Is this discussion going to at some point turn away from being a gripe session and into something productive? 

Or are we just going to keep flogging this point?

**Comment by wiztechinc** [comment]
both Crystal and posted views and suggestions.
ᐧ

Delia Wilson Lunsford
WizTech, Inc., (formerly Delia Wilson Design, LLC.)
434-202-4307
Terms and Conditions for working with WizTech, Inc.
(http://www.teamwiztech.com/terms-conditions.php)

**Comment by wiztechinc** [comment]
sorry both Crystal and I
ᐧ

Delia Wilson Lunsford
WizTech, Inc., (formerly Delia Wilson Design, LLC.)
434-202-4307
Terms and Conditions for working with WizTech, Inc.
(http://www.teamwiztech.com/terms-conditions.php)

**Comment by drbyte** [comment]
Okay, so you're making suggestions. Is that going to be the sum total of this discussion? You're going to tell us what you want to see us do? 

I thought @DivaVocals started out by asking whether actual code submissions were welcomed ... to which I said yes. But it seems we're back to simply saying Classic is ugly ... again.

**Comment by wiztechinc** [comment]
Um, I kinda thought this discussion had been nixed. I'll be glad to
elucidate further....
ᐧ

Delia Wilson Lunsford
WizTech, Inc., (formerly Delia Wilson Design, LLC.)
434-202-4307
Terms and Conditions for working with WizTech, Inc.
(http://www.teamwiztech.com/terms-conditions.php)

**Comment by DivaVocals** [comment]
@drbyte 
>Okay, on the issue of facelifting classic (the title of this discussion), all I'm seeing is everyone saying that Classic is ugly and needs to be changed. Is this discussion going to at some point turn away from being a gripe session and into something productive? 

Chris you are absolutely right.. Thanks for the kick in the butt here.. I intend to back this up with a pull request this weekend.. 

All this talk about the parent template (template_default) is NOT what the point of this issue. this is why I separated any discusssion about Classic from the parent template (template_default) issue I created.. My only point here is that Classic (the default CHILD template) needs a reboot and it doesn't need to wait while final decisions on the direction of the parent template (template_default) is decided..

Not sure what the BIG deal is.. :laughing: No one here REALLY uses Classic.. I don't and a number of you as much as admitted the same.. I am looking at this strictly from the POV of a newbie, DIY shopowner who may install Zen Cart today.. I daresay that some of us are really overthinking this.. :smile:

**Comment by wiztechinc** [comment]
I don't use the classic template as is but I do use it as the base so that
anyone who comes after me who knows the default template can figure out
what I've done. I feel that's part of the reason it exists. One of the
biggest problems with most bought templates is that nothing of the default
exists making some of them impossible to alter with any efficiency at all.

I think we need to look a bit further than just changing the css to make
the template more attractive - and it isn't just about "ugly". It's also
about a bit of visual modernizing and laying a base that gives a better
starting place.

The green classic template looks like zen cart isn't moving forward with
the rest of the world.

That's not a criticism. That's just an observation. How old is it? I can
barely remember the one before it.

I specified some changes earlier that I think is a direction that we need
to go in. Small things - just changing the blocky look of by using a
repeating background image on the body tag in order to give the appearance
of 100% width and reduce the amount of dead space on either side of the
actual container on really wide monitors. Or creating a header container
that is 100% wide for just header background color by making the
mainWrapper only the middle section of the page and not including the
header and footer. Just a little different approach that pays off big time.
ᐧ

Delia Wilson Lunsford
WizTech, Inc., (formerly Delia Wilson Design, LLC.)
434-202-4307
Terms and Conditions for working with WizTech, Inc.
(http://www.teamwiztech.com/terms-conditions.php)

**Comment by DivaVocals** [comment]
>I specified some changes earlier that I think is a direction that we need to go in. Small things - just changing the blocky look of by using a repeating background image on the body tag in order to give the appearance of 100% width and reduce the amount of dead space on either side of the actual container on really wide monitors. 

Suggest downloading and install Barebones.. It's already a 100% width template..

**Comment by wiztechinc** [comment]
don't like it
ᐧ

Delia Wilson Lunsford
WizTech, Inc., (formerly Delia Wilson Design, LLC.)
434-202-4307
Terms and Conditions for working with WizTech, Inc.
(http://www.teamwiztech.com/terms-conditions.php)

**Comment by DivaVocals** [comment]
That's fine.. but IMHO a default child template should be a minimalistic as possible..

**Comment by wiztechinc** [comment]
The look is fine. Used it once and never went back. Too much stuff had to
be changed and sorry I don't remember details. That was a long time ago.
ᐧ

Delia Wilson Lunsford
WizTech, Inc., (formerly Delia Wilson Design, LLC.)
434-202-4307
Terms and Conditions for working with WizTech, Inc.
(http://www.teamwiztech.com/terms-conditions.php)

**Comment by DivaVocals** [comment]
and as I tell my nephew and end-users, "don't like it" is not an answer.. :smile: I read your previous post, but honestly I am unclear what you are suggesting and none of your screenprints can be seen..

So let's start from Barebones as a starting point and let's here you suggestions fromthere.. Because I am not starting from the existing Classic template.. :smile:

**Comment by wiztechinc** [comment]
Well, that's telling me.
ᐧ

Delia Wilson Lunsford
WizTech, Inc., (formerly Delia Wilson Design, LLC.)
434-202-4307
Terms and Conditions for working with WizTech, Inc.
(http://www.teamwiztech.com/terms-conditions.php)

**Comment by DivaVocals** [comment]
Not telling you at all.. just saying let's start from a common point.. So far I don't have anything specific that I can comment on..

**Comment by wiztechinc** [comment]
Is github stripping out all urls? If so, we can't  give concrete examples to look at - and nothing I've said makes sense.

Delia Wilson Design  
delia@deliawilsondesign.com
(434) 202-1683

**Comment by drbyte** [comment]
When you post by email, attachments don't get included. But actual URLs in clear text (not html formatted <a href> links) are kept.

And, also, when you post by email, your "email signature" and all the "reply history" in the email also get posted to github, so would be tidier if you deleted all those when replying by email.

**Comment by DivaVocals** [comment]
>And, also, when you post by email, your "email signature" and all the "reply history" in the email also get posted to github, so would be tidier if you deleted all those when replying by email.

Yup... :smile:

That's why I stopped replying by email to Github.. PLUS the post by emails isn't reliable..

**Comment by wiztechinc** [comment]
Thanks drbyte for the info. So, let me try again. Everything I speak about here is shown as samples at http://deliatest.com/1_5_3  

These 4 things are what I think needs to be in the default template as an example for someone to use and based their own template on. This has nothing to do with whether it's the barebones, classic or timbuktu template. I think we should supply a usable template that only requires some minor changes to get a modern looking site.  I spent  time with this after taking a look at 1.6 and while trying to wrap my head around what's best for zen cart rather than what's best for me or even for my specific clients.  A different mindset as the core team members will tell you.

1. Instead of one container for the entire layout, I find that using three is more adaptable. header, content, and footer. If you know the classic template, the mainWrapper is the container with a fixed width.  Having the three sections (which is actually more HTML5 friendly) requires three containers, all with a fixed width. 

#header, #mainContent, #footer {
width: 960px, margin: 0 auto 0;
}

That allows for some visual effects. If this sounds elemental, just bear with me. I'm trying to be really clear - not assuming that anyone knows or doesn't know any of this. The first is an illusion that the site is 100% wide. Some of today's large monitors can leave a lot of space on either side of the main body. Of course 100% width does not work - it's then too wide. This solution mimics that standard bootstrap template:

body {
background-image url(../images/bk_body.jpg); background-repeat: x-repeat; background-position: 0 75px; 
}

The bk_body.jpg in this case is a bar maybe 30px high - similar to the tile_bk.gif of classic that then can be used for the top menu background.  If you want a different background color in the header than the body background, the image can be the full height. 

The advantage to this is that you can use a 1px wide repeating image which then loads fast. Yeah, I know. Duh.

You can do the same with the footer for 100% affect.

An alternative would be to have an outer container and then an inner container with background colors but I try to avoid adding nested containers for visual effects in order to reduce the amount of source code. 

2. Shopping cart in header. I've got the header sidebox ready to go. I think this should be part of the default, not a mod. The reason is simple but you have to look at your own shopping habits. Say you spend a fair amount of time deciding on your items on a site. The site gives free shipping if you buy $35. After about 30 minutes, two short conversations with the three year old, one robocall you go "wait, where was I? Did I put that book in the cart? What's my total up to?" Look to the header, no links, no clicking. "Aha, over $35!" Click on that link staring you in the face and buy. It's about ease of use and promoting sales. Most templates are including this in some way already.

3. The shopping cart description: I use a background image for a cart symbol. number of items, dollar amount and then the next line has the login and my account links. When there are items in the cart, the checkout link appears. This can be up top and in front or inline with the top menu links.  See this image: http://deliatest.com/1_5_3/images/cart_sideboxes.jpg  

4. Category boxes - make it an unordered list. Styling options are infinitely variable with a small change, reducing the need for an additional sidebox just for display purposes.

All of this is offered as examples. By putting our heads together I expect we can come up with even better solutions than I'm offering here. It's not just about code or css. It is about - once again - selling. Selling a potential zen cart user the cart to start with and selling that user's products.

**Comment by DivaVocals** [comment]
Let me make sure we are all on the same page and all using the right names..

**default_template** is the parent template in Zen Cart. As the name implies it IS the DEFAULT template..

Classic is NOT the default template.. Like any other Zen Cart template, Classic is a CHILD template. It is the CHILD template that comes bundled with Zen Cart. It is NOT the default template though.. 

Child templates inherit their layout from the default_template unless you use the overrides system to create a variation from the default.

Classic is a simple child template that consists of a stylesheet and some images.. It has always been a reflection of the parent template in terms of layout, but applies some images and color to give it a different look than the default template.

Delia much of what you suggest are changes that need to be part of a larger discussion on the future of the default template.. Otherwise putting that level of effort into a Classic template makeover without having a better sense as to the future of the underlying default templating system feels a lot like a waste of effort.. I was actually suggesting something a LOT simpler as a short term EASY change to improve the out the box look while the larger discussions on the future of the templating system start to take shape... Low hanging fruit with immediate benefit..

**Comment by barco57** [comment]
^what Diva said.
I was tempted to write a similar post but Diva does such a much better job, she knows exactly where to use ALL CAPS....:gift_heart:

**Comment by DivaVocals** [comment]
I wanna know how to do that heart so I can send one back to you Mike!!! :heart_decoration:

**Comment by drbyte** [comment]
In case @DivaVocals wasn't clear, Delia @wiztechinc raises good points, all worthy of consideration. Primarily for the default template.

It's true, this discussion is geared mainly around facelifting Classic.

**Comment by DivaVocals** [comment]
>Delia @wiztechinc raises good points, all worthy of consideration. Primarily for the default template.

Exactamundo!!! Any layout changes that are eventually made to the **new** default_template will become part of Classic as a it is child theme, it will inherit changes from the parent..  (notwithstanding that there of course may be some stylesheet changes needed to support the default template changes..)

**Comment by wiztechinc** [comment]
I got it. Crystal thumped me on the head to get it in there. Classic only.

**Comment by DivaVocals** [comment]
Thumped.. I thought I gave you a glass of wine first???? **giggle**

**Comment by wiztechinc** [comment]
That was last night. Today the thump

**Comment by DivaVocals** [comment]
:laughing:

**Comment by scottcwilson** [comment]
Now that we have the new responsive_classic in 1.5.5, should we close this issue?
